### ISSUE TITLE
Type of chain values are unknown when iterable cast to sequence

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1079]
**Describe the bug**
When creating a sequence directly from `itertools.chain`, i.e. without assigning the return value to a variable, the type of the sequence values is lost.  This issue first appeared in 1.1.76.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That `bar` would be of type `list[int]`.

**Screenshots or Code**
```python
# pyright: strict

from itertools import chain

foo = chain([0])
reveal_type(foo)  # Type of "foo" is "chain[int]"
bar = list(chain([0]))
reveal_type(bar)  # Type of "bar" is "list[Unknown]"
baz = list(foo)
reveal_type(baz)  # Type of "baz" is "list[int]"
```

**VS Code extension or command-line**
Extension, 1.1.76

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release. I've added a test case so this regression won't happen again.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.77, which I just published. It will also be included in the next release of Pylance.
